contracts/strategies/auraBAL/Strategy.sol

```diff
@@ -0,0 +1,202 @@
+    }
+
+    /// @notice Lock or swap 20WETH-80BAL LP tokens for auraBal
+    /// @param _lock - whether to lock or swap
+    /// @param _minAmountOut - minimum expected amount of auraBal
+    /// @return the amount of auraBal obtained from lock or swap
+    function _lockOrSwap(bool _lock, uint256 _minAmountOut)
+        internal
+        returns (uint256)
+    {
+        uint256 _bptBalance = IERC20(BAL_ETH_POOL_TOKEN).balanceOf(
+            address(this)
+        );
+        if (_lock) {
+            // if we lost to much too slippage, revert
+            if (_bptBalance < _minAmountOut) {
+                revert("slippage");
+            }
+            // lock without staking to receive auraBAL
+            bptDepositor.deposit(_bptBalance, true, address(0));
+            // the mint is 1:1
+            return _bptBalance;
+        } else {
+            // if not locking we'll swap on the pool
+            return _swapBptToAuraBal(_bptBalance, _minAmountOut);
+        }
+    }
+
+    /// @notice Claim rewards and swaps them to FXS for restaking
+    /// @dev Can be called by the vault only
+    /// @param _caller - the address calling the harvest on the vault
+    /// @param _minAmountOut -  min amount of LP tokens to receive w/o revert
+    /// @param _lock - whether to lock or swap
+    /// @return harvested - the amount harvested
+    function harvest(
+        address _caller,
+        uint256 _minAmountOut,
+        bool _lock
+    ) public onlyVault returns (uint256 harvested) {
+        // claim rewards
+        auraBalStaking.getReward();
+
+        // process rewards
+        address[] memory _rewardTokens = rewardTokens;
+        for (uint256 i; i < _rewardTokens.length; ++i) {
+            address _tokenHandler = rewardHandlers[_rewardTokens[i]];
+            if (_tokenHandler == address(0)) {
+                continue;
+            }
+            uint256 _tokenBalance = IERC20(_rewardTokens[i]).balanceOf(
+                address(this)
+            );
+            if (_tokenBalance > 0) {
+                IERC20(_rewardTokens[i]).safeTransfer(
+                    _tokenHandler,
+                    _tokenBalance
+                );
+                IRewardHandler(_tokenHandler).sell();
+            }
+        }
+
+        uint256 _wethBalance = IERC20(WETH_TOKEN).balanceOf(address(this));
+        uint256 _balBalance = IERC20(BAL_TOKEN).balanceOf(address(this));
+
+        // Deposit to BLP
+        _depositToBalEthPool(_balBalance, _wethBalance, 0);
+
+        // lock or swap the LP tokens for aura BAL
+        uint256 _auraBalBalance = _lockOrSwap(_lock, _minAmountOut);
+
+        uint256 _stakingAmount = _auraBalBalance;
+
+        if (_auraBalBalance > 0) {
+            // if this is the last call, no fees
+            if (IGenericVault(vault).totalSupply() != 0) {
+                // Deduce and pay out incentive to caller (not needed for final exit)
+                if (IGenericVault(vault).callIncentive() > 0) {
```

Comment:
You're calling `IGenericVault(vault).callIncentive()` and `IGenericVault(vault).platformFee()` twice in `Strategy.sol`. Putting it in a local variable might be a gas optimization?

Reply:
done